### PR TITLE
best-practice: support extra strategy for fine-tuning the performance in massive regions

### PR DESCRIPTION
**Pull request by qiancai**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.2 (TiDB 8.2 versions)
- [x] v8.1 (TiDB 8.1 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [x] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/17911
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: best-practices/massive-regions-best-practices.md [hunk 1]
```diff
@@ -51,6 +51,14 @@ You can check the following monitoring metrics in Grafana's **TiKV Dashboard**:
 
     ![Check Propose wait duration](/media/best-practices/propose-wait-duration.png)
 
++ `Commit log duration`  in the **Raft IO** panel
```
Comment (qiancai) [inline best-practices/massive-regions-best-practices.md:54]:
```suggestion
+ `Commit log duration` in the **Raft IO** panel
```

File: best-practices/massive-regions-best-practices.md [hunk 2]
```diff
@@ -136,6 +144,14 @@ The default size of a Region is 96 MiB, and you can reduce the number of Regions
 > + The query performance, especially for queries that deal with a large range of data, might decrease.
 > + The Region scheduling slows down.
 
+### Method 7: Increase the number of connections for Raft communication
+
+By default, the number of connections used for Raft communication between TiKV nodes is 1. Increasing this number can help alleviate blockage issues caused by heavy communication workloads of a large number of Regions. For detailed instructions, see [`grpc-raft-conn-num`](/tikv-configuration-file.md#grpc-raft-conn-num).
+
+> **Note:**
+>
+> To reduce unnecessary thread switching overhead and prevent negative impacts from batch processing, it is recommended to set the number within the range of `[1, 4]`.
```
Comment (LykxSassinator) [inline best-practices/massive-regions-best-practices.md:153]:
```suggestion
> To reduce unnecessary thread switching overhead and mitigate potential negative impacts from batch processing, it is recommended to set the number within the range of `[1, 4]`.
```

File: best-practices/massive-regions-best-practices.md [hunk 3]
```diff
@@ -51,6 +51,14 @@ You can check the following monitoring metrics in Grafana's **TiKV Dashboard**:
 
     ![Check Propose wait duration](/media/best-practices/propose-wait-duration.png)
 
++ `Commit log duration` in the **Raft IO** panel
+
+    `Commit log duration` is the time Raftstore takes to commit Raft logs to the majority of members in the respective Region. If this metric value is high and shows significant fluctuations, it indicates that the workload of Raftstore is heavy, the append log operation is slow, or Raft logs cannot be committed timely due to network congestion.
```
Comment (LykxSassinator) [inline best-practices/massive-regions-best-practices.md:56]:
```suggestion
    `Commit log duration` is the time Raftstore takes to commit Raft logs to the majority of members in the respective Region. A high value for this metric, with significant fluctuations, indicates that the workload on Raftstore is heavy, the append log operation is slow, or Raft logs cannot be committed timely due to network congestion..
```

**Review by LykxSassinator — COMMENTED** [review]
Rest LGTM

**Review by LykxSassinator — APPROVED** [review]

File: best-practices/massive-regions-best-practices.md [hunk 4]
```diff
@@ -51,6 +51,14 @@ You can check the following monitoring metrics in Grafana's **TiKV Dashboard**:
 
     ![Check Propose wait duration](/media/best-practices/propose-wait-duration.png)
 
++ `Commit log duration` in the **Raft IO** panel
+
+    `Commit log duration` is the time Raftstore takes to commit Raft logs to the majority of members in the respective Region. A high value for this metric, with significant fluctuations, indicates that the workload on Raftstore is heavy, the append log operation is slow, or Raft logs cannot be committed timely due to network congestion.
```
Comment (hfxsd) [inline best-practices/massive-regions-best-practices.md:56]:
```suggestion
    `Commit log duration` is the time Raftstore takes to commit Raft logs to the majority of members in the respective Region. A high value for this metric, with significant fluctuations, indicates that:
    
     - The workload on Raftstore is heavy.
     - The append log operation is slow.
     - Or Raft logs cannot be committed timely due to network congestion.
```

File: best-practices/massive-regions-best-practices.md [hunk 5]
```diff
@@ -136,6 +144,14 @@ The default size of a Region is 96 MiB, and you can reduce the number of Regions
 > + The query performance, especially for queries that deal with a large range of data, might decrease.
 > + The Region scheduling slows down.
 
+### Method 7: Increase the number of connections for Raft communication
```
Comment (hfxsd) [inline best-practices/massive-regions-best-practices.md:147]:
```suggestion
### Method 7: Increase the maximum number of connections for Raft communication
```

File: best-practices/massive-regions-best-practices.md [hunk 6]
```diff
@@ -136,6 +144,14 @@ The default size of a Region is 96 MiB, and you can reduce the number of Regions
...
+
+By default, the number of connections used for Raft communication between TiKV nodes is 1. Increasing this number can help alleviate blockage issues caused by heavy communication workloads of a large number of Regions. For detailed instructions, see [`grpc-raft-conn-num`](/tikv-configuration-file.md#grpc-raft-conn-num).
```
Comment (hfxsd) [inline best-practices/massive-regions-best-practices.md:149]:
```suggestion
By default, the maximum number of connections used for Raft communication between TiKV nodes is 1. Increasing this number can help alleviate blockage issues caused by heavy communication workloads of a large number of Regions. For detailed instructions, see [`grpc-raft-conn-num`](/tikv-configuration-file.md#grpc-raft-conn-num).
```

**Review by hfxsd — APPROVED** [review]

File: best-practices/massive-regions-best-practices.md [hunk 7]
```diff
@@ -51,6 +51,18 @@ You can check the following monitoring metrics in Grafana's **TiKV Dashboard**:
...
 
++ `Commit log duration` in the **Raft IO** panel
+
+    `Commit log duration` is the time Raftstore takes to commit Raft logs to the majority of members in the respective Region. The possible reasons for a high value of this metric with significant fluctuations include the following:
+    
+     - The workload on Raftstore is heavy.
+     - The append log operation is slow.
+     - Raft logs cannot be committed timely due to network congestion.
+
+    Reference value: lower than 200-500 ms.
```
Comment (qiancai) [inline best-practices/massive-regions-best-practices.md:62]:
```suggestion
  Reference value: lower than 200-500 ms.
```

File: best-practices/massive-regions-best-practices.md [hunk 8]
```diff
@@ -51,6 +51,18 @@ You can check the following monitoring metrics in Grafana's **TiKV Dashboard**:
 
     ![Check Propose wait duration](/media/best-practices/propose-wait-duration.png)
 
++ `Commit log duration` in the **Raft IO** panel
+
+    `Commit log duration` is the time Raftstore takes to commit Raft logs to the majority of members in the respective Region. The possible reasons for a high value of this metric with significant fluctuations include the following:
+
```
Comment (qiancai) [inline best-practices/massive-regions-best-practices.md:57]:
```suggestion

```

File: best-practices/massive-regions-best-practices.md [hunk 9]
```diff
@@ -51,6 +51,17 @@ You can check the following monitoring metrics in Grafana's **TiKV Dashboard**:
 
     ![Check Propose wait duration](/media/best-practices/propose-wait-duration.png)
 
++ `Commit log duration` in the **Raft IO** panel
+
+    `Commit log duration` is the time Raftstore takes to commit Raft logs to the majority of members in the respective Region. The possible reasons for a high value of this metric with significant fluctuations include the following:
+     - The workload on Raftstore is heavy.
+     - The append log operation is slow.
+     - Raft logs cannot be committed timely due to network congestion.
+
```
Comment (qiancai) [inline best-practices/massive-regions-best-practices.md:60]:
```suggestion
    `Commit log duration` is the time Raftstore takes to commit Raft logs to the majority of members in the respective Region. The possible reasons for a high value of this metric with significant fluctuations include the following:

    - The workload on Raftstore is heavy.
    - The append log operation is slow.
    - Raft logs cannot be committed timely due to network congestion.

```

**Pull request merged**